common/vrnetlab.py

```diff
@@ -408,7 +408,13 @@ def gen_mgmt(self):
         res = []
         # mgmt interface is special - we use qemu user mode network
         res.append("-device")
-        res.append(self.nic_type + f",netdev=p00,mac={gen_mac(0)}")
+        # we need to place mgmt interface on the same bus with other interfaces in OpenBSD,
+        # otherwise, it will be assigned the last index by the OS,
+        # and not the first (i.e., vio0) as desired
+        if "openbsd" in self.image:
+            res.append(self.nic_type + f",netdev=p00,mac={gen_mac(0)},bus=pci.1")
+        else:
+            res.append(self.nic_type + f",netdev=p00,mac={gen_mac(0)}")
```

Comment:
@dteslya do you mind instead of introducing the logic in common/vrnetlab.py to do this in the openbsd module instead?

You can call the parent's class gen_mgmt and then replace the element that adds nedtdev=p00 with the PCI bus you need to have.

Reply:
Thanks for the hint. Will try that.